services/web/server/src/simcore_service_webserver/projects/projects_utils.py

```diff
@@ -286,3 +294,69 @@ def any_node_inputs_changed(
                     )
                     return True
     return False
+
+
+def get_frontend_node_outputs_changes(
+    new_node: dict[str, Any], old_node: dict[str, Any], filter_keys: set[ServiceKey]
+) -> FrontendOutputsChanges:
+    """if node is a specific type it checks if outputs changed"""
+    nodes_keys = {old_node.get("key"), new_node.get("key")}
```

Comment:
THOUGHT: The function is very specific in the type of nodes it tackles i.e. "front-end nodes"
The implementation also expects some structure in ``new_node`` and ``old_node`` ``dict``s. 
Why not to express that in the type annotations and use ``TypedDict``s for them?